consensus/fbft_log.go

```diff
@@ -60,127 +56,179 @@ func (m *FBFTMessage) String() string {
 	)
 }
 
+// Hash return the hash of the FBFT message for the unique identifier for FBFT message
+func (m *FBFTMessage) Hash() common.Hash {
+	// TODO: verify with RJ whether this struct uniquely defines a FBFT message
+	hashData := struct {
+		MsgType   uint32
+		BlockHash common.Hash
+		SenderKey bls.SerializedPublicKey
+		LeaderKey bls.SerializedPublicKey
+		Payload   []byte
+	}{
+		MsgType:   uint32(m.MessageType),
+		BlockHash: m.BlockHash,
+		Payload:   m.Payload,
+	}
+	if m.SenderPubkey != nil {
+		hashData.SenderKey = m.SenderPubkey.Bytes
+	}
+	if m.LeaderPubkey != nil {
+		hashData.LeaderKey = m.LeaderPubkey.Bytes
+	}
+	return hash.FromRLPNew256(hashData)
```

Comment:
the rlp hash could be a performance bottleneck. better directly use some of the basic data fields such as blockHash+msgType+sender as key.

Reply:
Thanks for feedback. Updated with ID function to specify a unique FBFT message.